x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/InvalidMappedField.java

```diff
@@ -61,8 +54,8 @@ private InvalidMappedField(String name, String errorMessage, Map<String, EsField
         this.typesToIndices = typesToIndices;
     }
 
-    private InvalidMappedField(StreamInput in) throws IOException {
-        this(in.readString(), in.readString(), in.readImmutableMap(StreamInput::readString, i -> i.readNamedWriteable(EsField.class)));
+    public InvalidMappedField(StreamInput in) throws IOException {
```

Comment:
Does the ctor have to become public? I think we're ok keeping it private.

Reply:
Not really, I think I can make it protected at least